### PR TITLE
Implement charm loops as observables

### PR DESCRIPTION
**Pull request by mreboud**

### CODE REVIEW

File: eos/rare-b-decays/charm-loops.cc [hunk 1]
```diff
@@ -1476,6 +1479,195 @@ namespace eos
         return result;
     }
 
+
+    /*
+     * Adaptater that exports the charn loops function as observables
```
Comment (dvandyk) [inline eos/rare-b-decays/charm-loops.cc:1484]: "Adaptor"

File: eos/rare-b-decays/charm-loops.cc [hunk 2]
```diff
@@ -1476,6 +1479,195 @@ namespace eos
...
+     */
+    template <>
+    struct Implementation<CharmLoopsAdaptater>
```
Comment (dvandyk) [inline eos/rare-b-decays/charm-loops.cc:1487]: Adaptor :smile:

File: eos/rare-b-decays/charm-loops.cc [hunk 3]
```diff
@@ -1476,6 +1479,195 @@ namespace eos
...
+                 flag_ctQs = 0,
+                 flag_ctQb = 0;
+
+        Implementation(const Parameters & p, const Options & o, ParameterUser & u) :
+            opt_contribution(o, options, "contribution"),
+            m_b(p["mass::b(MSbar)"], u),
+            m_c(p["mass::c"], u),
+            mu(p["sb::mu"], u)
+        {
+            if ("0" == opt_contribution.value())
```
Comment (dvandyk) [inline eos/rare-b-decays/charm-loops.cc:1513]: This is better done with an ``std::map`` and ``std::tie``, isn't it?

```
static const std::map<std::string, std::tuple<double, double, double>> settings
{
    std::make_pair("all", std::make_tuple(1.0, 1.0, 1.0)),
    ...
};

auto i = settings.find(opt_contribution.value());
if (i == settings.end())
   throw ...;
 
// member variables
double flag_a, flag_b, ...;

std::tie(flag_a, flag_b, flac_) = i->second;
```

File: eos/rare-b-decays/charm-loops.cc [hunk 4]
```diff
@@ -1476,6 +1479,195 @@ namespace eos
...
+
+        inline agv_2019a::CharmLoopsParameters clp(const complex<double> & s) const
+        {
+            return agv_2019a::CharmLoopsParameters(mu / m_b, s / m_b() / m_b(), (m_c * m_c) / (m_b * m_b), 1e-12);
+        }
+
+        complex<double> F17(const complex<double> & s) const
+        {
+            auto params = clp(s);
+            return (double)flag_a * agv_2019a::f17a(params)
```
Comment (dvandyk) [inline eos/rare-b-decays/charm-loops.cc:1541]: I would declare the flags as ``double`` from the get go, to avoid the casting here.

**Review by dvandyk — CHANGES_REQUESTED** [review]

File: eos/rare-b-decays/charm-loops.cc (hunk 1 above)
Comment (mreboud) [reply eos/rare-b-decays/charm-loops.cc:1484]: Adapter? Like in form-factor-adapter?

**Pull request merged**